### ISSUE TITLE
Specify the config-schema rule that rejects ambiguous resource identity

### ISSUE THREAD

**Issue by MrBinnacle** [MrBinnacle/workspace_lint#19]
Split out of #16. `CONTEXT.md` now records the settled default — identity is the stable ID, names are report-only aliases — but the enforcing schema rule is implementation detail.

## What it enforces

A configuration entry identifying a resource by name alone is **rejected at load**, not resolved by search and not guessed at. A readable alias may accompany an ID and appears in reports.

## Why rejection rather than resolution

Resolving a name would make the scan's subject depend on a title any editor can change, and `docs/research/notion-api-documented.md` records that search is documented as non-exhaustive — *"Search is not guaranteed to return everything"*. A name-to-ID resolution step would therefore be both mutable and unreliable, and its failure mode is silent: the scan reads a different resource than the operator meant and reports coverage over it.

This is Principle 4 applied to configuration. The CLI infers nothing from labels alone.

## Definition of done

- The schema rule stated, with the rejection message an operator sees.
- The error names the offending entry and what to supply instead.
- A fixture config that fails to load, as the test.

## Revisit if

**Notion ships a stable, documented name-to-ID lookup that is exhaustive.** The rejection would then be avoidable strictness rather than a correctness guard. Search's non-exhaustiveness is the entire basis for this rule.

**Operators report the ID requirement as the reason they abandon setup.** `PRODUCT.md` already carries first-run configuration cost as a known risk and `docs/research/solo-segment-evidence.md` names the config file as the suspect. If this rule is a measurable share of that cost, the answer is a helper that resolves names **once, at config-authoring time, writing IDs into the file** — never a resolution at scan time.

**Comment by MrBinnacle** [comment]
## Folded in: this issue is now the build ticket for the config schema, not only the spec

Two build tickets were filed for the remaining v0.1 rules — `REQ001` and `UNQ001`, the two **Configured** entries in `CONTEXT.md`'s rule catalog. Both are blocked here, so the scope of this issue is worth restating against what the code now does.

**Half of this issue's definition of done is already shipped.** `slice/config.ts` on `build/t3-ref001` implements the rejection for declared roots: a `NAME_KEYS` list (`name`, `title`, `page`, `url`, `link`), a distinct message for the named-but-unidentified case, and the citation to `CONTEXT.md`'s settled default. It returns a reason and the caller exits `4`, on the grounds that an invalid configuration means the scan never began.

What is **not** built is the thing the two new tickets need: there is nowhere in the schema to configure a rule. `Config` carries `version`, `roots` and `minCoverage`. A required property (`REQ001`) and a uniqueness scope with a value selector (`UNQ001`) have no home.

**Revised scope for this issue:**

- [ ] The schema gains a validated rule-configuration section.
- [ ] The identity rule already enforced for `roots` is enforced there too — a rule configured against a resource named by title alone is rejected at load, never resolved by search and never guessed at.
- [ ] The rejection names the offending entry and what to supply instead, matching the message shape `config.ts` already uses.
- [ ] A fixture config that fails to load is the test.
- [ ] Rejection exits `4` (ADR-0008 decision 2 — the scan did not run as declared).

**Unchanged:** the reason for rejecting rather than resolving. `docs/research/notion-api-documented.md` records that search is documented as non-exhaustive, so a name-to-ID step would be both mutable and unreliable, and its failure mode is silent — the scan reads a different resource than the operator meant and reports coverage over it. That is Principle 4 applied to configuration.

**Both Revisit-ifs on the original body stand**, including the one that matters most for adoption: if the ID requirement turns out to be a measurable share of first-run configuration cost, the answer is a helper that resolves names **once, at config-authoring time, writing IDs into the file** — never a resolution at scan time.

**Comment by MrBinnacle** [comment]
## Half of this ticket already ships on `main`

Raised by an external review of the board and verified against the file.

`slice/config.ts:65-74` already rejects a root addressed by name alone, with this ticket's own rationale in the comment above it and the message this ticket's Definition of Done asks for:

```
roots[i] is addressed by <key> and carries no "id". Identity is the stable ID;
a name is a report-only alias and is never resolved to a resource ...
```

It covers `name`, `title`, `page`, `url`, `link` and `alias` via `NAME_KEYS`. So the first two DoD lines — *"the schema rule stated, with the rejection message an operator sees"* and *"the error names the offending entry and what to supply instead"* — are done. Whether a fixture config that fails to load exists as a **test** was not checked here.

## What remains, and it is the load-bearing half

**The rule-configuration surface.** `slice/config.ts` validates `version`, `roots` and `minCoverage`. There is nowhere to declare a required property or a uniqueness scope, which is what **#58 (`REQ001`)** and **#59 (`UNQ001`)** consume — #58's body already says so under "Blocked by".

**Suggested restatement of the title:** *"Extend the config schema with the rule-configuration section"*, with the name-rejection half recorded as shipped. The current title describes the finished half, which makes this ticket read as untouched when it is roughly half done.

**Not doing that rename here** — it changes what the ticket is, and #58 and #59 both point at it. Flagging for the operator.

**Comment by MrBinnacle** [comment]
## The load-bearing half is built — `b6518e6` on `build/19-rule-config-schema`

The revised scope on this ticket, line by line.

- [x] **The schema gains a validated rule-configuration section.** `rules`, optional, empty when
      absent — a config without it behaves exactly as before.
- [x] **The identity rule enforced for `roots` is enforced there too.** It is now ONE
      implementation, `parseResourceRef`, called for declared roots and rule scopes alike. It was
      about to be two, and the second copy is the one that gets corrected without the first.
- [x] **The rejection names the offending entry and what to supply instead**, matching the message
      shape `config.ts` already used — `rules[0].scope is addressed by alias and carries no "id"…`.
- [x] **A fixture config that fails to load is the test.** `slice/CHECK-config.ts`, 64 assertions.
- [x] **Rejection exits `4`** (ADR-0008 decision 2). Verified end to end through the CLI, which
      rejects before any call is made.

`REQ001` is the section's only member. #70 decision 1 — whether `UNQ001` is Configured, built-in or
split — is open, and a schema that guessed at `UNQ001`'s shape would be rewritten by whichever way
that decision falls. `UNQ001` is rejected at load with a message naming #59.

## What the loader refuses, and why each one is a refusal rather than a default

| Input | Refused because |
| --- | --- |
| A scope addressed by name, alias or URL | Identity is the stable ID. Search is documented non-exhaustive, so resolution would be mutable, unreliable and silent. |
| A `REQ001` entry with no `scope` | A required-property rule with no scope asserts the property over every enumerated resource — inference of applicability from nothing, which ADR-0001 decision 4 rejects. |
| A rule ID outside the catalog | A typo that loads is a rule nobody configured. |
| `SYS001` / `REF001` | Built-in. They take no configuration and run on every scan. |
| `UNQ001`, `SCH001`, `REL001`, `DEP001`, `CAN001` | Catalogued, not built. Two messages, not one: an operator who typed `REQ002` has a different problem from one who configured `UNQ001` early. |
| A duplicate `(rule, normalized scope ID, property)` | `REQ001`'s coverage item is (resource, property) PAIRS — ADR-0011. A duplicate adds a pair to the denominator that no observation ever fills. |

## The hole the example config opened

Writing `wl.config.example.json` exposed a defect in the design above. **The loader validates the
document against a hand-kept table**, so it said yes to `REQ001` while the scan runs
`[SYS001, REF001]` and evaluates nothing the operator declared — a green run over a rule that never
ran, in this product's own entry point.

Closed by asking the **build** rather than the table: `unimplementedRules(configured, implemented)`
in `rule.ts`, wired into `cli.ts` before any call, with `BUILT_RULES` exported from `scan.ts` so the
list has one definition. That export also deleted a hand-written scalar — the CLI header printed
`rules implemented: 2 (SYS001, REF001)` as a literal.

## Three defects found by review, all introduced by this change

- `RULE_STATUS[id]` resolved through `Object.prototype`. `{"rule":"toString"}` walked past the
  not-in-catalog branch with `status.kind` undefined and told the operator their typo was a real
  catalog rule shipping later. `Object.hasOwn` now; the four prototype twins are asserted.
- `property` was validated with `.trim()` and **stored raw**, so `"Owner"` and `"Owner "` were two
  entries — the duplicate this ticket's own test exists to close, one field to the right.
- A present-but-**malformed** `id` got the identity message, telling an operator who had written an
  `id` to supply an `id`. It fired on the likeliest path there is, since the worked example pairs an
  `id` with an `alias`.

Nothing in the suite loaded `wl.config.example.json`; TEST 8 does now. The example shipped broken in
the first version of this change — a `REQ001` entry the CLI rejected at exit 4 — and no assertion
said so.

## Controls

- Bypass the scope identity rejection → `CHECK-config.ts` exits 1. Restored, exits 0.
- Remove `tsx CHECK-config.ts` from the check chain → `CHECK-suite-registration.ts` exits 1 naming
  the file. Restored, exits 0.
- The claim gate went **red before the scalars were updated** — `checkpoint.md:219 count — 10
  file(s) match slice/CHECK-*.ts`, exit 1. Counterfactual run, not argued.
- Both scored on the **exit code**. A crashed suite prints no `FAIL`.

`cd slice && npm run check` — **676 assertions, ten suites, offline, exit 0.**

## What this does NOT close

**#58 keeps its other blocking edge.** #18, the rule-to-hydration map, is untouched by this and is
now `REQ001`'s only remaining blocker.

**#78 is new and is not blocked by anything.** `property` is validated as a non-empty string and
nothing more. A Notion property carries a stable `id` as well as a mutable `name`, so the identity
argument reaches it — but property IDs are opaque and the Notion UI does not surface them, which is a
usability objection the resource rule never faced. It needs two live-API facts, not an argument.

**Recommended: close this ticket.** Both halves of its Definition of Done ship. The rename suggested
in the comment above — *"Extend the config schema with the rule-configuration section"* — is moot if
it closes.
